### PR TITLE
feat: get contacts that are not part of the conversation (AR-1699)

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- in order to add the participants to a group, we first need to get all the known users that are not part of the converation

### Solutions

- use the existing getConversationMembers() function and combine it with getAllKnownUsers() inside the UseCase to filter the AllKnownUsers with the one that are part of the conversation

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

- Unit tests

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by yamilmedina — COMMENTED** [review]
Great work as usual, master! 🚀 Added some questions and suggestions, let me know what you think 😄

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+) {
+    suspend operator fun invoke(conversationId: QualifiedID) =
+        conversationRepository.getConversationMembers(conversationId).flatMap { conversationMembers ->
+            userRepository.getAllContacts()
+                .map { allContacts -> removeConversationMembersFromAllContacts(allContacts, conversationMembers) }
+        }.fold({ Result.Failure(it) }, { Result.Success(it) })
+
+    private fun removeConversationMembersFromAllContacts(allContacts: List<OtherUser>, conversationMembers: List<UserId>) =
+        allContacts.filter { conversationMember -> !conversationMembers.contains(conversationMember.id) }
+}
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt:25]: i wonder if doing this on the sql level will be faster than iterating over 2 whole lists to filter.

Comment (trOnk12) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt:25]: Valid Point ill look into

Comment (trOnk12) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt:25]: will make a change and submit the change soon

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+sealed class Result {
+    data class Success(val contactNotInTheConversation: List<OtherUser>) : Result()
+    data class Failure(val storageFailure: StorageFailure) : Result()
+}
```
[the remark this answers is not in the document]
Comment (trOnk12) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInTheConversationUseCase.kt]: actually it was not internal forgot to expose it

**Review by MohamadJaara — APPROVED** [review]
lgtm 🚀

**Pull request merged**